### ISSUE TITLE
Cube width bug in away 2.3.

### ISSUE THREAD

**Issue by pdelia** [pdelia/away3d#212]
#32 Issue by __GoogleCodeExporter__, created on: 2015-04-24T07:51:14Z

```
When I change width or height of the cube I get TypeError: Error #1009: at
away3d.primitives::Cube/buildCube()

Here is a code that reproduces this bug:

import away3d.containers.*;
import away3d.core.base.*;
import away3d.primitives.*;
import away3d.core.math.*;

var view:View3D = new View3D({x:300, y:200});

addChild(view);

var cube:Cube = new Cube({material:"blue#", name:"cube", x: 300, y:
160, z: -80, width:200, height:200, depth:200});

view.scene.addChild(cube);

view.camera.position = new Number3D(1000, 1000, 1000);
view.camera.lookAt(cube.position);
cube.width=1000; //HERE GOES AN ERROR

view.render(); 
```

Original issue reported on code.google.com by `antonkul...@gmail.com` on 13 Feb 2009 at 11:35

Attachments:
* [cubeWidthBug_in_2_3.fla](https://storage.googleapis.com/google-code-attachments/away3d/issue-32/comment-0/cubeWidthBug_in_2_3.fla)

**Comment by pdelia** [comment]
Comment by __GoogleCodeExporter__, created on: 2015-04-24T07:51:15Z

```
I have fixed this bug, but I'm not sure if this fix wouldn't cause new bugs. 
Problem
was in Geometry.as in removeFace function. When cube do buildPrimitive() in
AbstractPrimitive.buildPrimitive() faces are removing. But after that in 
buildCube()
it cause an error because _backFaces[i].uv0 is null as well as others uv. To 
fix this
bug I simply remove this lines of code in Geometry.as in removeFace
face.uv0 = null;
face.uv1 = null;
face.uv2 = null;
```

Original comment by `karmazin...@gmail.com` on 13 Feb 2009 at 12:42

**Comment by pdelia** [comment]
Comment by __GoogleCodeExporter__, created on: 2015-04-24T07:51:15Z

```
fixed in revison 1111, version 2.3.1
```

Original comment by `rob.bate...@gmail.com` on 17 Feb 2009 at 1:01
* Changed state: **Fixed**
